homeassistant/components/http.py

```diff
@@ -185,6 +195,41 @@ def register_path(self, method, url, callback, require_auth=True):
         """ Regitsters a path wit the server. """
         self.paths.append((method, url, callback, require_auth))
 
+    def check_expired_sessions(self):
+        """ Reemove any expired sessions. """
+        if SESSION_LOCK.acquire(False):
+            try:
+                keys = []
+                for key in self._sessions.keys():
+                    keys.append(key)
+
+                for key in keys:
+                    if self._sessions[key].is_expired:
+                        del self._sessions[key]
+                        _LOGGER.info("Cleared expired session %s", key)
+            finally:
+                SESSION_LOCK.release()
+
+    def add_session(self, key, session):
+        """ Add a new session to the list of tracked sessions """
+        self.check_expired_sessions()
+
+        if not SESSION_LOCK.acquire():
```

Comment:
Calling blocking without argument is blocking. It should not happen that this returns `False`